src/main/java/org/broadinstitute/hellbender/engine/MultiVariantDataSource.java

```diff
@@ -94,8 +95,8 @@ public MultiVariantDataSource(final List<FeatureInput<VariantContext>> featureIn
         // 2) Create and cache a merged header using versions of the individual headers from each data source that
         //    have been updated to include the actual dictionary returned from that data source
         //
-        validateAllSequenceDictionaries();
-        mergedHeader = getMergedHeader();
+        validateAllSequenceDictionaries(errorOnOutOfDateIndex);
+        mergedHeader = getMergedHeader(errorOnOutOfDateIndex);
```

Comment:
Check for out-of-date index should happen in the constructors, not in the sequence dictionary validation.